test/test_datasets.py

```diff
@@ -490,6 +490,89 @@ def inject_fake_data(self, tmpdir, config):
         return num_images_per_category * len(categories)
 
 
+# @mock.patch('torchvision.datasets.imagenet._verify_archive')
+class ImageNetTestCase(datasets_utils.ImageDatasetTestCase):
+    DATASET_CLASS = datasets.ImageNet
+    REQUIRED_PACKAGES = ['scipy']
+
+    def imagenet_root(self, root):
```

Comment:
@pmeier this is still WIP but I might need your advice here:

I ported this from the current `test/fakedata_generation.py`, but this thing generates `tar` archives instead of generating raw files. In a normal setting, the tar are supposed to be extracted in the `ImageNet` constructor.

Applying the mocks in `datasets_utils` above will unfortunately prevent the extraction of the tar, which is why I commented them out (for now)

Would you suggest that I:

a) find a way to properly extract those `tar` in `ImageNetTestCase.ImageNetTestCase`? (one hacky way is to literally call `torchvision.datasets.ImageNet(tmpdir, split='train')`, but that's not great)
b) not build the `tar` and completely change `imagenet_root()` to create the raw files instead?

Reply:
> b) not build the tar and completely change `imagenet_root()` to create the raw files instead?

This is the way to go. Before in some cases we provided the archives while sometimes we provided the raw data. We agreed that the download / extract logic is fairly easy to review manually, so we opted to always provide the raw data. This makes the implementation of `inject_fake_data` a lot easier because we don't need to mess with archives.

This is not documented and this is on me. I'll send a PR to address this.